src/pt.erl

```diff
@@ -0,0 +1,132 @@
+-module(pt).
+
+-export([parse_transform/2]).
+
+parse_transform(Forms, Options) ->
+    % io:fwrite("Forms = ~p~n", [Forms]),
+    % Forms.
+    {seed, Seed} = lists:keyfind(seed, 1, Options),
+    Trans = fun (Form) -> do_transform(Seed, Form) end,
+    parse_trans:plain_transform(Trans, Forms).
+
+
+do_transform(_, {'op', L, '!', Lhs, Rhs}) ->
+    [NewRhs] = parse_trans:plain_transform(fun transform_send/1, [Rhs]),
+    {'op', L, '!', Lhs, NewRhs};
+do_transform(Seed, {'receive', _, _} = T) ->
+    Trans = fun (Form) -> transform_receive(Seed, Form) end,
+    hd(parse_trans:plain_transform(Trans, [T]));
+do_transform(_, _) ->
+    continue.
+
+
+% In case of sending a message, what we want is to send a message as is, but also
+% to log it somehow (e.g.: we are printing it out in this example)
+% Because the RHS of ! doesn't allow sequencing with ',', we have to somehow wrap the
+% print statement and the expressin we are sending, so they will be represented by a
+% single syntactic unit
+% That can be done by just having them as the body of a lambda, which is then applied immediately
+% Pid ! {self(), hello}
+% Pid ! (fun() ->
+%                io:format("~p~n",[{self(), hello}]),
+%                {self(), hello}
+%        end)()
+
+transform_send(T) ->
+    L = erl_syntax:get_pos(T),
+    {call,
+     L,
+     {'fun',
+      L,
+      {clauses,
+       [{clause,
+         L,
+         [],[],
+         [{call,
+           L,
+           {remote,L,{atom,L,io},{atom,L,format}},
+           [{string,L,"Sent: ~p~n"},{cons,L,T,{nil,L}}]}, % This is where we log the message
+          T % This is the original message
+         ]}]}},
+     []}.
```

Comment:
It would be nice to use [syntax tree construction functions](https://github.com/erlang/otp/blob/master/lib/syntax_tools/src/erl_syntax.erl) instead of hardcoding the tree representation.